projects/igniteui-angular/src/lib/grids/cell.component.ts

```diff
@@ -885,11 +887,13 @@ export class IgxGridCellComponent implements OnInit, OnChanges, OnDestroy {
     }
 
     private setActiveNode() {
+        const currRow = this.grid.dataView[this.rowIndex];
+        const type: GridKeydownTargetType = this.grid.isDetailRecord(currRow) ? 'masterDetailRow' : 'dataCell';
         if (this.grid.navigation.activeNode) {
-            Object.assign(this.grid.navigation.activeNode, {row: this.rowIndex, column: this.visibleColumnIndex});
+            this.grid.navigation.setActiveNode({ row: this.rowIndex, column: this.visibleColumnIndex}, type);
         } else {
             const layout = this.column.columnLayoutChild ? this.grid.navigation.layout(this.visibleColumnIndex) : null;
-            this.grid.navigation.activeNode = { row: this.rowIndex, column: this.visibleColumnIndex, layout: layout };
+            this.grid.navigation.setActiveNode({ row: this.rowIndex, column: this.visibleColumnIndex, layout: layout }, type);
```

Comment:
is this if clause still necessary ... since in both statements we use setActiveNode method and if any additional logic is necessary is better to be moved inside setActiveNode method.